### PR TITLE
added index to tracks and trackstags

### PR DESCRIPTION
**Pull request by gferraro**
changed recording query to query Tags and/or TracksTags tables when relevant.
also changed tests to add trackstags for animals instead of recording tags

### CODE REVIEW

File: migrations/20191111002157-track-index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+module.exports = {
+  up: async function(queryInterface) {
+    await queryInterface.addIndex("Tracks", ["RecordingId"]);
+    await queryInterface.addIndex("TrackTags", ["TrackId"]);
+  },
+  down: async function(queryInterface) {
+    await queryInterface.removeIndex("Tracks", ["RecordingId"]);
+    await queryInterface.removeIndex("TrackTags", ["TrackId"]);
+  }
+};
```
Comment (mjs) [inline migrations/20191111002157-track-index.js]: so these make a decent difference to the query plan?

File: models/Recording.js [hunk 2]
```diff
@@ -580,6 +580,21 @@ module.exports = function(sequelize, DataTypes) {
 
     const humanSQL = 'NOT "Tags".automatic';
     const AISQL = '"Tags".automatic';
+    if (models.Tag.acceptableTags.includes(tagMode)) {
```
Comment (mjs) [inline models/Recording.js]: Is this right? It doesn't look like `tagMode` would ever be in `acceptableTags`.

File: migrations/20191111002157-track-index.js (hunk 1 above)
Comment (gferraro) [reply migrations/20191111002157-track-index.js]: yep it uses these indexes which remove a cost  a cost of 20% due to searching via a hash prior

File: models/Recording.js (hunk 2 above)
Comment (gferraro) [reply models/Recording.js]: when you serach on browse for cool that is put into the tagMode. I think we should change this so we sepearte recording Tags and TrackTags in the query

Comment (mjs) [reply models/Recording.js]: Separating the track and recording tags in the query makes a lot of sense.

**Pull request merged**